src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/MetricsEmitter.java

```diff
@@ -847,4 +848,141 @@ public static void emitNodeMetrics(
           "Total time taken for writing {} metrics metricsdb: {}", tableName, mFinalT - mCurrT);
     }
   }
+
+    public static void emitFaultDetectionMetrics(DSLContext create, MetricsDB db, FaultDetectionMetricsSnapshot faultDetectionSnapshot) {
+
+      long mCurrT = System.currentTimeMillis();
+      Dimensions dimensions = new Dimensions();
+      Result<Record> res = faultDetectionSnapshot.fetchAggregatedTable();
+      LOG.error("AHH record size is " + res.size());
+      List<String> dims =
+              new ArrayList<String>() {
+                {
+                  this.add(AllMetrics.FaultDetectionDimension.SOURCE_NODE_ID.toString());
+                  this.add(AllMetrics.FaultDetectionDimension.TARGET_NODE_ID.toString());
+                }
+              };
```

Comment:
Redundant ? We can use the `dimensions` defined on line 855

Reply:
This is used to store a key value. Dimension is a map object that has key as metric dimension and value as to what is the value for it whereas dims is a list of dimension as string . I have still created arraylist from dimension only now